### ISSUE TITLE
Doesn't reset properly when switching from items to loadOptions

### ISSUE THREAD

**Issue by itay-grudev** [rob-balfre/svelte-select#111]
I have two modes and I need to toggle between them:
* Single select item with a static list of options
* Multiple select with a dynamic list of options

Currently switching between the two mostly works, with exception to some minor issue like this:

**Expected:**

0. The select is set to `Single`
1. I switch to `Multi`
2. Nothing really happens

**Actual:**

0. The select is set to `Single`
1. I switch to `Multi`
2. The items from the `Single` mode appear.

[REPL](https://svelte.dev/repl/f899b2630d2e49b8b75a26b09a2be053?version=3.12.1).

**Comment by artanik** [comment]
For multi select `selectedValue` option should be as an array, so to get it done you need to make something like `selectOptions.selectedValue = [selectedValue];` when `mode == 'MULTI'`.

[REPL](https://svelte.dev/repl/9500856e0e024b9bac89a8be4f33cb0d?version=3.12.1)

**Comment by rob-balfre** [comment]
@artanik thanks

**Comment by itay-grudev** [comment]
@rob-balfre Please reopen it, it's still an issue. You just misunderstood the problem.
Based on @artanik's code I just added the following:

```js
if (mode == 'MULTI') {
  selectOptions.items = [];
  selectOptions.selectedValue = [];
  selectOptions.isMulti = true;
  selectOptions.loadOptions = loadOptions;
}
```

The idea is that when the field is switched to Multi we are not going to use the predefined set of items but totally different one loaded dynamically. For this reason I set `items = []` and `selectedValue = []` as the old value is no longer valid.

What happens is that whenever I switch to `MULTI`:
1. The item list is automatically displayed
2. The old `items` persist and the new items are not shown.

Furthermore if I set `items` in `MULTI` to:

```js
selectOptions.items = [
  {value: 'broccoli', label: '🥦 Broccoli', group: 'Sweet'}
];

It never appears.
```

The core issue I think is that `items` doesn't get updated correctly in this situation.

**Comment by artanik** [comment]
@itay-grudev sorry for the misunderstanding. You are right. In this case, if a component already has `items` and these `items` were "cleared" (or not), then `loadOptions` not fired at all but also an "open" state was triggered.

**Comment by itay-grudev** [comment]
@rob-balfre @artanik This issue is blocking my work. If there is any way I can assist you please feel free to ping me.

**Comment by itay-grudev** [comment]
It would take me significant amount of effort to reverse-engineer the code and fix it myself, but if you don't have the time ti deal with it now that is an acceptable option, of course, but please say so.

**Comment by artanik** [comment]
Well, I do not participate in this project. This issue was suggested by GitHub, and it looked easy to fix when I first saw it. But then it turned out that it is an internal issue and my knowledge is limited here. I have to reverse-engineer the code as well. You can make a PR if you're confident in your fix or just wait for the author's fix.

**Comment by rob-balfre** [comment]
@itay-grudev https://svelte.dev/repl/e978e97fb2694d67a4775475498fb7ac?version=3.12.1

**Comment by itay-grudev** [comment]
@rob-balfre If I am not mistaken, this solution bypasses the issue as it re-initialises the whole `Select` component instead of updating it's options.

Thank you - I'll use it and it saves my ass. But... Wouldn't it be better if the source of the problem is eliminated as well?

**Comment by rob-balfre** [comment]
@itay-grudev yeah, I'm a bit low on time so thought a bypass is better than nothing!  I'll keep the issue open and fix it up when I can or feel free to raise a PR.

**Comment by rob-balfre** [comment]
fixed in v4.0.0-beta.11

https://svelte.dev/repl/3e55a06dc86e421fa1579ec87aa9340b?version=3.12.1
